internal/services/pam/pam_test.go

```diff
@@ -324,18 +324,27 @@ func TestIsAuthenticated(t *testing.T) {
 		"Successfully authenticate":                           {username: "success"},
 		"Successfully authenticate if first call is canceled": {username: "IA_second_call", secondCall: true, cancelFirstCall: true},
 		"Denies authentication when broker times out":         {username: "IA_timeout"},
+		"No error when auth.Next and no data":                 {username: "IA_next"},
 		"Update existing DB on success":                       {username: "success", existingDB: "cache-with-user.db"},
 
 		// service errors
 		"Error when sessionID is empty": {sessionID: "-"},
 		"Error when there is no broker": {sessionID: "invalid-session"},
 
 		// broker errors
-		"Error when authenticating":                         {username: "IA_error"},
-		"Error on empty data even if granted":               {username: "IA_empty_data"},
-		"Error when broker returns invalid access":          {username: "IA_invalid"},
-		"Error when broker returns invalid data":            {username: "IA_invalid_data"},
-		"Error when calling second time without cancelling": {username: "IA_second_call", secondCall: true},
+		"Error when authenticating":                               {username: "IA_error"},
```

Comment:
So, let’s try to simplify this. The errors here are not triggered directly by the PAM module but by the brokers. So let’s ensure we have one way to trigger the isAuthenticated from the broker failing, and report something around it. Otherwise, we will have for a change to update 2 places and 2 tests failing and so on (actually, we already had too many tests then).

There is still the second call case, and some validation that is done by the pam service to keep.